packages/superset-ui-preset-chart-xy/src/encodeable/types/data.ts

```diff
@@ -0,0 +1,8 @@
+export interface PlainObject {
+  [key: string]: any;
+}
+
+export interface Dataset {
+  keys: string[];
+  values: PlainObject[];
```

Comment:
not sure if we can make sure that the key in the `PlainObject` will be always in the set of `key`? seems impossible.\

Reply:
I googled a bit and couldn't figure this out, worth asking in #typescript tho?

Reply:
Can use typeguard to help. Not if possible to define solely as type without knowing the fields before hand.

```ts
export type PlainObject<Key extends string = string, Value extends any = any> = {
  [key in Key]: Value
};

export type Dataset<T extends string> = {
  keys: T[];
  values: Partial<PlainObject<T>>[];
}

function isDataset<T extends string>(input: Dataset<T>): Dataset<T> {
  return input;
}
```

![image](https://user-images.githubusercontent.com/1659771/54395257-97e09180-466c-11e9-9fa5-61c6e651dc04.png)

Reply:
![image](https://user-images.githubusercontent.com/1659771/54395399-e8f08580-466c-11e9-80df-fed21e9d6bf1.png)

Reply:
mmm. this is more complicated. the guard only work for inline input, but if you pass a variable with type already it make `keys` `string[]` instead of `'a'|'b'|'c'[]`